lib/kafka_ex/protocol/common.ex

```diff
@@ -0,0 +1,16 @@
+defmodule KafkaEx.Protocol.Common do
+  @moduledoc """
+  A collection of common request generation and response parsing functions for the
+  Kafka wire protocol.
+  """
+
+  @doc """
+  Generate the wire representation for a list of topics.
+  """
+  def topic_data([]), do: ""
+
+  def topic_data([topic|topics]) do
```

Comment:
So, also just for the sake of conversation, I think "cleaner" is subjective here.  For whatever reason, when I learned Erlang, I got "use tail call recursion" drilled into my head.  So I'm used to thinking that way and seeing something that's NOT tcr is jarring to me ;)  Which is just to say: there are different points of view on the topic.